src/devices/Servo/ServoMotor.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Device.Gpio;
+using System.Device.Pwm;
+using System.Device.Pwm.Drivers;
+using System.IO;
+
+namespace Iot.Device.Servo
+{
+    public class ServoMotor : IDisposable
+    {
+        private int _servoPin;
+        private int _pwmChannel;
+        private double _pulseFrequency = 20.0;
+        private double _angle;
+        double _currentPulseWidth = 0;
+        /// <summary>
+        /// Are we running on hardware or software PWM? True for hardware, false for sofware
+        /// </summary>
+        public bool IsRunningHardwarePwm { get; set; }
+
+        /// <summary>
+        /// Servo motor definition.
+        /// </summary>
+        private ServoMotorDefinition _definition;
+        /// <summary>
+        /// The duration per angle's degree.
+        /// </summary>
+        private double _rangePerDegree;
+        private PwmController _pwmController;
+
+        /// <summary>
+        /// Initialize a ServoMotor class
+        /// </summary>
+        /// <param name="pinNumber">The GPIO pin number in case of a software PWM. The chip in case of a hardware PWM</param>
+        /// <param name="pwmChannel">The channel to use in case of a hardware PWM.</param>
+        /// <param name="definition">The definition of a ServoMotor</param>
+        /// <remarks>Use -1 for pwmChannel to force using software PWM</remarks>
+        public ServoMotor(int pinNumber, int pwmChannel, ServoMotorDefinition definition)
+        {
+            this._definition = definition;
+            _pulseFrequency = definition.PeriodMicroseconds / 1000.0;
+
+            UpdateRange();
+
+            _servoPin = pinNumber;
+            this._pwmChannel = pwmChannel;
+            // try hardware PWM first
+            try
+            {
+                _pwmController = new PwmController();
+                _pwmController.OpenChannel(pinNumber, pwmChannel);
+                IsRunningHardwarePwm = true;
+
+            }
+            catch (Exception ex)
```

Comment:
optional: this construct you wrote here can be done with `catch (Exception ex) when (ex is ArgumentException || ex is IOException)`